mesonbuild/compilers.py

```diff
@@ -2729,3 +2733,135 @@ def unix_compile_flags_to_native(self, args):
 
     def get_link_debugfile_args(self, targetfile):
         return []
+
+
+# Only tested on linux for ICC 17.0.0
+#    * object extraction does not compile (don't know) (25 object extraction fails)
```

Comment:
Quite possibly; the depfile contains: `main 1@exe/main.c.o: ../test\ cases/common/25\ object\ extraction/main.c`. The dependency is right, but the space in the target is not escaped, and that's probably not right.